cli/src/main/java/software/amazon/smithy/java/cli/SmithyCall.java

```diff
@@ -0,0 +1,253 @@
+        for (String smithyFile : BASE_RESOURCE_FILES) {
+            URL resourceUrl = classLoader.getResource(smithyFile);
+            if (resourceUrl != null) {
+                assembler.addImport(resourceUrl);
+            } else {
+                System.err.println("Resource not found: " + smithyFile);
+            }
+        }
+
+        // Add model files
+        try {
+            assembler.addImport(directoryPath);
+            return assembler.assemble().unwrap();
+        } catch (Exception e) {
+            throw new IOException("Failed to assemble model from directory: " + e, e);
+        }
+    }
+
+    private ShapeId validateServiceExists(Model model) {
+        ShapeId serviceInput = ShapeId.from(service);
+        if (!model.getShapeIds().contains(serviceInput)) {
+            throw new IllegalArgumentException("Service " + service + " not found in model");
+        }
+        return serviceInput;
+    }
+
+    private DynamicClient buildDynamicClient(Model model, ShapeId serviceInput) {
+        if (url == null) {
+            throw new IllegalArgumentException("Service endpoint URL is required. Please provide the --url option.");
+        }
+
+        DynamicClient.Builder builder = DynamicClient.builder()
+                .service(serviceInput)
+                .model(model)
+                .endpointResolver(EndpointResolver.staticEndpoint(url));
+
+        configureAuth(builder, serviceInput);
+        configureProtocol(builder, serviceInput);
+        configureInputInterceptor(builder);
+
+        return builder.build();
+    }
+
+    private void configureAuth(DynamicClient.Builder builder, ShapeId serviceInput) {
+        String defaultArnNamespace = serviceInput.getNamespace().toLowerCase();
+        if (authType != null) {
+            switch (authType.toLowerCase()) {
+                case "sigv4":
+                case "aws":
+                    if (awsRegion == null) {
+                        throw new IllegalArgumentException("SigV4 auth requires --aws-region to be set. Please provide the --aws-region option.");
+                    }
+                    builder.putConfig(RegionSetting.REGION, awsRegion)
+                            .putSupportedAuthSchemes(new SigV4AuthScheme(defaultArnNamespace))
+                            .authSchemeResolver(AuthSchemeResolver.DEFAULT)
+                            .addIdentityResolver(new EnvironmentVariableIdentityResolver());
+                    break;
+                case "none":
+                    builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported auth type: " + authType);
+            }
+        } else {
+            builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+        }
+    }
+
+    private void configureProtocol(DynamicClient.Builder builder, ShapeId serviceInput) {
```

Comment:
It would be cool if a custom build of the CLI could register additional protocols and auth schemes.